### ISSUE TITLE
Returning object in function is bugged when used with spread operator

### ISSUE THREAD

**Issue by melvin2016** [Microsoft/TypeScript#29431]
**TypeScript Version:**  3.3.0-dev.201xxxxx

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
//interface 
interface bagConfig {
    zip?: boolean;
    size: number;
    name: string;
    [propname: string]:any
   };
   
function showBagDet(bag1: bagConfig): {name:string,size:number,zip?:boolean}{
    console.log(`Bag name : ${bag1.name}\nBag Size: ${bag1.size} l\nDoes it have zip: ${bag1.zip===true?"Yes":"No"}`);
    return{
        name:bag1.name,
        size:bag1.size,
        ...bag1
    };
}

const bag = showBagDet({zip:true,size:5.6,name:"American Tourister",opasity:0.5,htl:123} as bagConfig);
console.log(bag);
```

**Expected behavior:**
the function ``` showbagDet() ``` only has 3 types to be returned but returns every keys in the object when spread operator used.
**Actual behavior:**
Typescript should raise an error while doing so .

**Comment by RyanCavanaugh** [comment]
This is the intended behavior. The object created as a result of the spread is a subtype of the return type.
